10up/retro-winamp-block
# Add feat: make winamp's window order configurable; Prevent windows from moving arbitrarily

### Description of the Change



Closes #10 (https://github.com/10up/retro-winamp-block/issues/10#issuecomment-1982053604)
Closes #83 

Added configuration items for winamp block to set the windows' displaying order. 

![image](https://github.com/10up/retro-winamp-block/assets/24775856/6c967040-bb41-444c-8fe7-148a34d985c7)

The Wordpress blocks can be placed into post/page, as well as sidebar widget. I've set rules separately. If you have any suggestions about these rules, just feel free to leave comments.

- Main window is forcedly displayed.
- All windows are NOT able to be dragged arbitrarily. 
- The windows are displayed by the order of configuration only for general viewing mode (not for post/page editing mode)

#### winamp block in post/page

![image](https://github.com/10up/retro-winamp-block/assets/24775856/29427ca2-273c-43ac-96f8-c953cdefb8b4)

#### winamp block in sidebar widget

![无标题](https://github.com/10up/retro-winamp-block/assets/24775856/199c46ce-0611-47f9-8c2c-58d70579237f)

### How to test the Change


1. Install the plugin
2. Add a winamp block in post/page/sidebar widget
3. Load media file so that the player can display
    - The winamp's windows cannot be dragged arbitrarily
5. In block setting tab, configure EQUALIZER/PLAYLIST/MILKDROP 's position as you wish
    - The position's change **won't** happen right now in editor mode
6. Click "Publish"/"Update" to save the post/page, or save the sidebar widget
7. Preview the post/page, the winamp windows should align according to the configuration.

> [!NOTE]  
> Known issue before this PR: If two or more blocks appear in the same webpage, only **one** winamp instance will be created.
>
> ![image](https://github.com/10up/retro-winamp-block/assets/24775856/71f3f0ec-8879-43ad-b704-df605e90c7b8)

### Changelog Entry

> Added - New feature: added configuration items for winamp block to set the windows' displaying order.
> Fixed - Bug fix: added javascript code to prevent winamp windows from unwanted moving.


### Credits

Props @dejalavidavolar


### Checklist:
